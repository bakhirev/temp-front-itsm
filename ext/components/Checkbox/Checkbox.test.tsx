import React from 'react';
import { shallow } from 'enzyme';

import { Checkbox } from './';

jest.mock('../common', () => ({
  DEFAULT_THEME: {},
}));

describe('Checkbox', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const checkboxRequiredProps = {
    onChange: jest.fn(),
  };

  it('should render component with required properties', () => {
    const wrapper = shallow(<Checkbox {...checkboxRequiredProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with label', () => {
    const wrapper = shallow(<Checkbox {...checkboxRequiredProps} label="Checkbox" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with small size', () => {
    const wrapper = shallow(<Checkbox {...checkboxRequiredProps} size={'small'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with indeterminate state', () => {
    const wrapper = shallow(<Checkbox {...checkboxRequiredProps} indeterminate />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClick when use clicks on component', () => {
    const wrapper = shallow(<Checkbox {...checkboxRequiredProps} />);
    wrapper.simulate('click');
    const { onChange } = checkboxRequiredProps;
    expect(onChange).toBeCalledTimes(1);
  });
});
