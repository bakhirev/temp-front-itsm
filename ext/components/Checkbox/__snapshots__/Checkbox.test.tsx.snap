// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render component with indeterminate state 1`] = `
<styled.div
  onClick={[Function]}
  onKeyUp={[Function]}
  size="big"
  tabIndex={1}
  theme={Object {}}
>
  <styled.input
    checked={false}
    readOnly={true}
    theme={Object {}}
    type="checkbox"
  />
  <styled.div
    size="big"
    tabIndex={-1}
    theme={Object {}}
    withLabel={false}
  >
    <Component />
  </styled.div>
</styled.div>
`;

exports[`Checkbox should render component with label 1`] = `
<styled.div
  onClick={[Function]}
  onKeyUp={[Function]}
  size="big"
  tabIndex={1}
  theme={Object {}}
>
  <styled.input
    checked={false}
    readOnly={true}
    theme={Object {}}
    type="checkbox"
  />
  <styled.div
    checked={false}
    size="big"
    tabIndex={-1}
    theme={Object {}}
    withLabel={true}
  >
    <Component />
  </styled.div>
  <Styled(styled.div)
    theme={Object {}}
  >
    Checkbox
  </Styled(styled.div)>
</styled.div>
`;

exports[`Checkbox should render component with required properties 1`] = `
<styled.div
  onClick={[Function]}
  onKeyUp={[Function]}
  size="big"
  tabIndex={1}
  theme={Object {}}
>
  <styled.input
    checked={false}
    readOnly={true}
    theme={Object {}}
    type="checkbox"
  />
  <styled.div
    checked={false}
    size="big"
    tabIndex={-1}
    theme={Object {}}
    withLabel={false}
  >
    <Component />
  </styled.div>
</styled.div>
`;

exports[`Checkbox should render component with small size 1`] = `
<styled.div
  onClick={[Function]}
  onKeyUp={[Function]}
  size="small"
  tabIndex={1}
  theme={Object {}}
>
  <styled.input
    checked={false}
    readOnly={true}
    theme={Object {}}
    type="checkbox"
  />
  <styled.div
    checked={false}
    size="small"
    tabIndex={-1}
    theme={Object {}}
    withLabel={false}
  >
    <Component />
  </styled.div>
</styled.div>
`;
